apps/server/src/modules/user/entity/user.entity.ts

```diff
@@ -0,0 +1,45 @@
+import { Entity, Property, Index, Unique } from '@mikro-orm/core';
+import { BaseEntityWithTimestamps, EntityId } from '@shared/domain';
+import { Role } from './role.entity';
+
+export interface IUserProperties {
+	email: string;
+	firstName?: string;
+	lastName?: string;
+	roles: Role[];
+	school: EntityId;
+}
+
+@Entity({ tableName: 'users' })
+export class User extends BaseEntityWithTimestamps {
+	@Property()
+	@Index({ name: 'externalUserIdentifier' })
+	@Unique()
+	email: string;
+
+	@Property()
+	firstName?: string;
+
+	@Property()
+	lastName?: string;
+
+	@Index({ name: 'roleIdBasedSearches' })
+	// @ManyToMany({ fieldName: 'roles', type: Role })
+	// roles = new Collection<Role>(this);
+	@Property()
+	roles: EntityId[] = [];
+
+	// index
+	// collection
+	@Index({ name: 'searchUserForSchool' })
```

Comment:
Was haben die folgenden Bezeichnungen gemeinsam?
"primaryRoleSearchOperation"
"externalUserIdentifier"
"roleIdBasedSearches"
"searchUserForSchool"

Vielleicht sollte es irgendeine Namenskonvention für indexe geben, oder zumindest irgendeine Form von Konsistenz, zb. ob sich der Name auf das Attribut oder die Operation bezieht, ob wir singular oder plural benutzen, ...

Reply:
Momentan ist das eher eine Erklärung wofür die ich im Code und in der DB hinterlegen möchte. Wie ein kurzer Stichpunkt ohne Leerzeichen. Wenn man kein Name setzt, wird ein default verwendet (zumindest in mongoose und hier denke ich auch.) 
Aber es ist extrem schwer von den default Namensschema auf den use case zu schließen. 

Aber sollten wir uns dafür entscheiden von default auf Namenskonvention zu wechseln (was ich begrüßen würde), sollte sie den use case Beschreiben.

Reply:
Falls du im Kontext der Sichtweise ein Vorschlag hast, oder kurz brainstormen möchtest, können wir das sofort einarbeiten.